src/hotspot/share/code/codeCache.cpp

```diff
@@ -1559,6 +1559,34 @@ void CodeCache::log_state(outputStream* st) {
             unallocated_capacity());
 }
 
+void CodeCache::write_perf_map(outputStream* st) {
```

Comment:
Could this whole function possibly live inside os/linux in an own file? Or would that expose too many code heap internals?

Reply:
Probably creates too much dependency between the os layer and the codeCache internals? I'll put it all in `#ifdef LINUX` though.